src/renderer/caching_shaper.rs

```diff
@@ -48,101 +40,159 @@ struct Asset;
 
 const DEFAULT_FONT_SIZE: f32 = 14.0;
 
+#[derive(Clone)]
+pub struct ExtendedFontFamily {
+    pub fonts: Vec<SkriboFont>,
+}
+
+impl ExtendedFontFamily {
+    pub fn new() -> ExtendedFontFamily {
+        ExtendedFontFamily { fonts: Vec::new() }
+    }
+
+    pub fn add_font(&mut self, font: SkriboFont) {
+        self.fonts.push(font);
+    }
+
+    pub fn get(&self, props: Properties) -> Option<&Font> {
+        for handle in &self.fonts {
+            let font = &handle.font;
+            let properties = font.properties();
+
+            if properties.weight == props.weight && properties.style == props.style {
+                return Some(&font);
+            }
+        }
+
+        if let Some(handle) = &self.fonts.first() {
+            return Some(&handle.font);
+        }
+
+        None
+    }
+
+    pub fn to_family(self) -> FontFamily {
+        let mut new_family = FontFamily::new();
+
+        for font in self.fonts {
+            new_family.add_font(font);
+        }
+
+        new_family
+    }
+}
+
+pub struct FontLoader {
+    cache: LruCache<String, ExtendedFontFamily>,
+    source: SystemSource,
+}
+
+impl FontLoader {
+    pub fn new() -> FontLoader {
+        FontLoader {
+            cache: LruCache::new(10),
+            source: SystemSource::new(),
+        }
+    }
+
+    fn get(&mut self, font_name: &str) -> Option<ExtendedFontFamily> {
+        if let Some(family) = self.cache.get(&String::from(font_name)) {
```

Comment:
This can be simplified with `cloned` https://doc.rust-lang.org/std/option/enum.Option.html#method.cloned

Reply:
sweet